sound/soc/sof/ipc3-topology.c

```diff
@@ -149,6 +173,74 @@ static void sof_ipc3_widget_free_comp(struct snd_sof_widget *swidget)
 	kfree(swidget->private);
 }
 
+static int sof_ipc3_widget_setup_comp_pipeline(struct snd_sof_widget *swidget)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct sof_ipc_pipe_new *pipeline;
+	struct snd_sof_widget *comp_swidget;
+	size_t ipc_size = sizeof(*pipeline);
+	int ret;
+
+	pipeline = (struct sof_ipc_pipe_new *)
+	       sof_comp_alloc(swidget, &ipc_size, swidget->pipeline_id);
+	if (!pipeline)
+		return -ENOMEM;
+
+	swidget->private = pipeline;
+
+	/* configure dai IPC message */
+	pipeline->hdr.size = sizeof(*pipeline);
```

Comment:
Hrm, if you use `sof_comp_alloc()` then the size has been already set (pipeline_id and comp_id as well), but _if_ we added the UUID in there we just ignore it?
Show we `kzalloc()` here only? If not then we should not change the size and no need to set pipeline_id and comp_id.

Reply:
on good catch. I should not modify the size. but I do need to set the pipeline_Id.

Reply:
You call 'sof_comp_alloc()' with swidget->pipeline_id and you set the pipeline_id to the same few lines later.

And this is not a comp message, it is a tplg one? I might be wrong, but I think this is kzalloc instead sof_comp_alloc ?

Reply:
fixed now